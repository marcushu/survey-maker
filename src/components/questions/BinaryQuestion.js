import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Form from 'react-bootstrap/Form';
import './questionStyles.css'


export default function BinaryQuestion({ question, updateQuestion }) {
  const [selected, setselected] = useState();

  const handleChange = e => {
    setselected(e.target.value);
    question.result = e.target.value;
    updateQuestion(question);
  }

  return (
    <div className="listItem d-flex justify-content-between" style={{border: "none"}}>
      <div className="question">
        {question.question}
      </div>
      <div>
        <Form.Check
          inline
          type='radio'
          label={question.positiveIdentity.replace('_', ' ')}
          value={question.positiveIdentity}
          checked={selected === question.positiveIdentity}
          onChange={e => handleChange(e)}
        />
        <Form.Check
          inline
          type='radio'
          label={question.negativeIdentity.replace('_', ' ')}
          value={question.negativeIdentity}
          checked={selected === question.negativeIdentity}
          onChange={e => handleChange(e)}
        />
      </div>
    </div>
  )
}

BinaryQuestion.propTypes = {
  question: PropTypes.object,
  updateQuestion: PropTypes.func
}