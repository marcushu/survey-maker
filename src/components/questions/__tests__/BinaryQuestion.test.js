import React from 'react';
import {  render } from '@testing-library/react';
import BinaryQuestion from '../BinaryQuestion';

describe("BinaryQuestion", () => {
    const mockQuestion = {
        negativeIdentity: "no",
        positiveIdentity: "yes",
        question: "mock question?"
    };

    const mockfunc = jest.fn();


    it("renders", () => {
        const { asFragment } = render(<BinaryQuestion
            question={mockQuestion}
            updateQuestion={mockfunc} />
            );

        expect(asFragment()).toMatchSnapshot();
    })

    it("correctly renders the labels (yes/no)", () => {
        const { getByDisplayValue } = render(<BinaryQuestion
            question={mockQuestion}
            updateQuestion={mockfunc} />
            );

        expect(getByDisplayValue(/yes/i)).toBeInTheDocument();
        expect(getByDisplayValue(/no/i)).toBeInTheDocument();
    })


})