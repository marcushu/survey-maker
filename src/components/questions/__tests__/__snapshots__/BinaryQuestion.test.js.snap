// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BinaryQuestion renders 1`] = `
<DocumentFragment>
  <div
    class="listItem d-flex justify-content-between"
  >
    <div
      class="question"
    >
      mock question?
    </div>
    <div>
      <div
        class="form-check form-check-inline"
      >
        <input
          class="form-check-input"
          type="radio"
          value="yes"
        />
        <label
          class="form-check-label"
          title=""
        >
          yes
        </label>
      </div>
      <div
        class="form-check form-check-inline"
      >
        <input
          class="form-check-input"
          type="radio"
          value="no"
        />
        <label
          class="form-check-label"
          title=""
        >
          no
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
